internal/controlplane/handlers_repositories.go

```diff
@@ -165,13 +165,13 @@ func (s *Server) ListRepositories(ctx context.Context,
 		repoId = sql.NullInt64{Valid: true, Int64: reqRepoCursor.RepoId}
 	}
 
-	limit := sql.NullInt32{Valid: false, Int32: 0}
+	limit := sql.NullInt64{Valid: false, Int64: 0}
 	reqLimit := in.GetLimit()
 	if reqLimit > 0 {
 		if reqLimit > maxFetchLimit {
 			return nil, util.UserVisibleError(codes.InvalidArgument, "limit too high, max is %d", maxFetchLimit)
 		}
-		limit = sql.NullInt32{Valid: true, Int32: reqLimit + 1}
+		limit = sql.NullInt64{Valid: true, Int64: reqLimit + 1}
```

Comment:
Does this mean that a user who doesn't pass in `limit` gets an unlimited list, but if they pass in a `limit`, we limit them to 100?

It feels like we should _always_ enforce a limit, so maybe the following code would be simpler?

```golang
limit := maxFetchLimit
if in.GetLimit() > 0 && in.getLimit < maxFetchLimit {
	limit = in.GetLimit()
}
```

And then later on, we could just check `if len(repos) == limit` on line 204.

Reply:
I agree, but the minder CLI doesn't support paginating results. `minder repo list` is why we check that way.